src/components/new-safe/create/useSyncSafeCreationStep.ts

```diff
@@ -20,8 +20,8 @@ const useSyncSafeCreationStep = (setStep: StepRenderProps<NewSafeFormData>['setS
       return
     }
 
-    // Jump to connect wallet step if there is no wallet and no pending Safe
-    if (!wallet) {
+    // Jump to the welcome page if there is no wallet and no pending Safe
+    if (!wallet && !pendingSafe) {
```

Comment:
I think this also doesn't fix the original issue when reloading the page since both wallet and pendingSafe are undefined at the start so the user always gets redirected back.